packages/react/src/components/List/ListItem/ListItem.jsx

```diff
@@ -39,7 +39,14 @@ const ListItemPropTypes = {
   expanded: PropTypes.bool,
   value: PropTypes.string.isRequired,
   /** string value or callback render function */
-  secondaryValue: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  secondaryValue: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.func,
+    PropTypes.shape({
```

Comment:
You will also have to update the file `packages/react/src/components/List/ListPropTypes.js` which is used by the List component.